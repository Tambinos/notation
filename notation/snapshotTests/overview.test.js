import React from "react";
import { render } from "@testing-library/react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import OverviewScreen from "../app/overview.tsx";

jest.mock("expo-router", () => {
	const React = require("react");
	return {
		useRouter: () => ({ push: jest.fn() }),
		useFocusEffect: (cb) => {
			React.useEffect(cb, []); // run after render
		},
	};
});

jest.mock("expo-sharing", () => ({
	isAvailableAsync: jest.fn().mockResolvedValue(true),
	shareAsync: jest.fn(),
}));

jest.mock("expo-file-system", () => ({
	File: class {
		constructor(_path, filename) {
			this.uri = `file:///${filename}`;
		}
		write = jest.fn();
	},
	Paths: { cache: "/cache" },
}));

jest.mock("../utils/AsyncStorage", () => ({
	getAllItems: jest.fn().mockResolvedValue({}),
	removeItem: jest.fn(),
}));

jest.mock("../app/tile-renderer.tsx", () => {
	const { View } = require("react-native");
	return function MockTileRenderer() {
		return <View testID="mock-tile-renderer" />;
	};
});

describe("OverviewScreen", () => {
	it("renders correctly and matches snapshot", async () => {
		const tree = render(
			<SafeAreaProvider>
			<OverviewScreen />
			</SafeAreaProvider>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});

