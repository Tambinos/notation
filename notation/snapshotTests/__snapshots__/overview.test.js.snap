// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewScreen renders correctly and matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
